### ISSUE TITLE
imputing 1-order derivative confounds

### ISSUE THREAD

**Issue by jdkent** [poldracklab/fitlins#212]
ref #56

Since fmriprep also calculates derivatives (which result in an `n/a` as the first value), should the first row of those columns also be imputed?

**Comment by adelavega** [comment]
Seems reasonable. I think this should apply to all variables that have an `n/a` in this first row because a value cannot be calculated.

**Comment by effigies** [comment]
The reason we didn't do it for all variables is that it's not clear that it's appropriate for all variables. Do we have a heuristic for identifying first-order derivatives? IIRC fMRIPrep uses a non-compliant naming scheme.

Or we could just say first TR is a good enough heuristic.

**Comment by adelavega** [comment]
Yeah, I'm not sure if we should go for all variables actually, although that would be the reasonable heuristic. 

I meant more that I think any variables from fmriprep to which this applies should be added to the list of variables we do this for.

**Comment by effigies** [comment]
Yeah, I guess, though I'd prefer if we stuck to the [draft standard](https://bids-specification.readthedocs.io/en/derivatives/05-derivatives/05-functional-derivatives.html#time-series-transformations), rather than fMRIPrep's custom names.

See https://github.com/poldracklab/fmriprep/issues/1864

**Comment by jdkent** [comment]
we could put it on the user to specify which confounds they want imputed through a command line option. Separate but related issue, we could let the user specify what confounds they want with a regular expression which helps confound columns that vary between participants (e.g., `motion_outlier[0-9]{2}`) which could be used to specify which confounds to impute as well.

[I have a solution in nibetaseries](https://github.com/HBClab/NiBetaSeries/blob/04a679c164de1316efbac67a289a9ea33d46f64f/src/nibetaseries/interfaces/nistats.py#L298-L314) for the regex expression.

**Comment by effigies** [comment]
For that I would probably prefer to go with simple globs like `*_derivative1`, to match stats models rules.

But sure, I think allowing users to specify imputation by CLI option probably makes sense, as long as it's not supported by the spec (which it's not).
